cargo-dist/src/config/v1/mod.rs

```diff
@@ -481,6 +483,173 @@ impl TomlLayer {
             }
         }
     }
+
+    /// Determines whether the configured install paths are compatible with each other
+    pub fn validate_install_paths(&self) -> DistResult<()> {
+        if let Some(installers) = &self.installers {
+            if let Some(paths) = &installers.common.install_path {
+                if paths.len() > 1 && paths.contains(&InstallPathStrategy::CargoHome) {
+                    return Err(DistError::IncompatibleInstallPathConfiguration {});
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    // We say "value is being ignored", but it seems like we use the "ignored" values anyway?
+    // However, for the v0->v1 migration, I'm going for reproducing behavior
+    // over fixing bugs, so I'm keeping it this way for now. -@duckinator
+    fn merge_warn(name: &str, package_manifest_path: &Utf8Path) {
+        warn!("dist.{} is set, but this is only accepted in workspace.metadata (value is being ignored): {}", name, package_manifest_path);
+    }
+
+    /// Merge a workspace config into a package config (self)
+    pub fn merge_workspace_config(
+        &mut self,
+        workspace_config: &Self,
+        package_manifest_path: &Utf8Path,
+    ) {
+        // This is intentionally written awkwardly to make you update it
```

Comment:
do we think this "awkwardness" strategy is still useful? (may still be worth not changing atm but i am curious your thoughts)

Reply:
I think there's probably a better way to do it, but I'm not sure what that is. But it _has_ actively caught problems for me a few times, so I think until we find a proper alternative, it's better than not destructuring it.